File: easy/problems/0441-arranging-coins/solution.py
def arrangeCoins(n):
    """
    Determine how many complete stairs can be built with n coins.

    Args:
        n: int - number of coins

    Returns:
        int - number of complete stairs
    """
    left, right = 0, n

    while left <= right:
        mid = (left + right) // 2
        cost = mid * (mid + 1) // 2

        if cost == n:
            return mid
        elif cost < n:
            left = mid + 1
        else:
            right = mid - 1

    return right

File: easy/problems/0441-arranging-coins/solution.cpp
class Solution {
public:
    int arrangeCoins(int n) {
        long left = 0, right = n;

        while (left <= right) {
            long mid = (left + right) / 2;
            long cost = mid * (mid + 1) / 2;

            if (cost == n) {
                return mid;
            } else if (cost < n) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return right;
    }
};
